examples/src/main/java/org/bitcoinj/examples/ForwardingService.java

```diff
@@ -23,34 +24,37 @@
 import org.bitcoinj.base.Coin;
 import org.bitcoinj.core.Context;
 import org.bitcoinj.core.InsufficientMoneyException;
-import org.bitcoinj.core.LegacyAddress;
 import org.bitcoinj.core.NetworkParameters;
+import org.bitcoinj.core.Transaction;
+import org.bitcoinj.core.TransactionConfidence;
 import org.bitcoinj.crypto.KeyCrypterException;
 import org.bitcoinj.kits.WalletAppKit;
 import org.bitcoinj.utils.BriefLogFormatter;
 import org.bitcoinj.wallet.KeyChainGroupStructure;
 import org.bitcoinj.wallet.SendRequest;
 import org.bitcoinj.wallet.Wallet;
+import org.bitcoinj.wallet.listeners.WalletCoinsReceivedEventListener;
 
 import java.io.File;
-
-import static com.google.common.base.Preconditions.checkNotNull;
+import java.io.PrintStream;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
 
 /**
  * ForwardingService demonstrates basic usage of the library. It sits on the network and when it receives coins, simply
  * sends them onwards to an address given on the command line.
+ * TODO: Needs testing
  */
 public class ForwardingService {
     static final String usage = "Usage: address-to-send-back-to [mainnet|testnet|signet|regtest]";
     static final int requiredConfirmations = 1;
-    private final BitcoinNetwork network;
-    private final NetworkParameters params;
-    private final Address forwardingAddress;
     private final WalletAppKit kit;
+    private Wallet wallet;
 
-    public static void main(String[] args) throws Exception {
+    public static void main(String[] args) {
         // This line makes the log output more compact and easily read, especially when using the JDK log adapter.
         BriefLogFormatter.init();
+        PrintStream out = System.out;
```

Comment:
I'm not a fan of caching `System.out` like that, and (further down) pass it into methods. `System.out` is a global variable, and it has always been used like that.

If we decide to go for that style, we should apply it to the entire framework (where is applies, possibly only examples/tools).

Reply:
We can drop this for now (and probably permanently.)

For the record, I like it for three reasons:

1. `System.out.println` gets verbose when used on many lines.
2. If you pass a `PrintStream` parameter to a method it allows that method to also write to other stream types.
3. In my refactoring I could use my `out` variable to easily check that all print output had been moved to a single method.